hibernate-core/src/main/java/org/hibernate/cfg/annotations/BasicValueBinder.java

```diff
@@ -204,7 +205,22 @@ public EnumType getEnumeratedType() {
 
 	@Override
 	public boolean isLob() {
-		return isLob;
+		if ( isLob ) {
+			return true;
+		}
+
+		if ( explicitJdbcTypeAccess != null ) {
+			final JdbcType type = explicitJdbcTypeAccess.apply( getTypeConfiguration() );
+			if ( type != null ) {
+				switch ( type.getJdbcTypeCode() ) {
+					case SqlTypes.BLOB:
+					case SqlTypes.CLOB:
+					case SqlTypes.NCLOB:
+						return true;
+				}
```

Comment:
How about this instead?

```suggestion
				return type.isLob();
```